canonical/loki-k8s-operator
# Alertmanager Integration

## Issue
Implement integration between Alertmanager and Loki, so any alerting rules _actually_ fire alerts.

## Solution
Primarily this was simply additions to `metadata.yaml` and small changes to the charm. The bulk of it was the creation of a workload-less tester charm which can generate messages in an arbitrary format through Juju actions.

## Testing Instructions
Deploy Loki and Alertmanager, as well as some charm which includes alerting rules (or `cos-config` with rules which match). See if alerts fire.

## Release Notes
Integrate Loki and Alertmanager, so that alerts fired by Loki based on LogQL alert rules are sent to Alertmanager for routing.
